src/Platforms/MySQL/CharsetMetadataProvider/CachingCharsetMetadataProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Platforms\MySQL\CharsetMetadataProvider;
+
+use Doctrine\DBAL\Platforms\MySQL\CharsetMetadataProvider;
+
+use function array_key_exists;
+
+/**
+ * @internal
+ */
+final class CachingCharsetMetadataProvider implements CharsetMetadataProvider
+{
+    private CharsetMetadataProvider $charsetMetadataProvider;
```

Comment:
You can use constructor property promotion here and below (maybe also with the `readonly` keyword?)

Reply:
👍🏻  for CPP. 🙂 

> (maybe also with the `readonly` keyword?)

We would need to bump to PHP 8.1 for that.

Reply:
It would be nice to enforce constructor property promotion in the coding standard. Otherwise, all code merged up from older branches will remain not using this syntax.

Reply:
https://github.com/doctrine/coding-standard/pull/283